Automattic/wp-calypso
# Jetpack: Action and data-layer for changing connection owner

Adds a redux action and data layer handlers for calling the Jetpack REST API to change the connection owner. The new API was added to Jetpack 6.2 in [this PR](https://github.com/Automattic/jetpack/pull/9578).

Part of larger PR #24876 that adds UI for changing the connection owner.

No visual differences in this PR.

## Testing
* Check out and build this branch
* Use a dotorg site with jetpack 6.2
* Connect the site to your main dotcom account
* Add another dotorg admin user to the dotorg site
* Connect the new user to a different dotcom account

Now the site has two admins, connected to different wpcom accounts. The first connected admin is the "Connection owner", and will be able to use the API call added in this PR to switch ownership to the other user.

**Testing a failing call**
* Go to http://calypso.localhost:3000/ in a browser, logged in as the first user (connection owner)
* In the JS console use this command:
`dispatch( { type: 'JETPACK_CONNECTION_CHANGE_OWNER', siteId: {site id}, newOwnerDotorgId: 9999, newOwnerWpcomDisplayName: 'blah blah' } )`
* Note the dotorg user id here is 9999. This should fail and show a notice. You can also view the failure in the network tab.

**Testing a successful call**
* Use the same command, but substitute the dotorg user id of the admin that does not own the connection. (To get the user IDs, you can go to https://developer.wordpress.com/docs/api/console/ and use the call `/sites/{site}/users`)
* You should see a success notice
* You will not be able to change the user again from this acount. You would need to log out and back in as the wpcom user that now owns the connection.

**Testing with UI**
* Follow test instructions in #24876
